### DISCUSSION TITLE
Livewire v3 multiple components on page

### DISCUSSION THREAD

**Discussion by Jeremiah-salley** [livewire/livewire#6997] [1. Help]
On one of our dashboard pages, a datatable is used to display the data. In the datatables 'actions' column are two raw html buttons that onclick trigger a javascript function that'll emit to one of two livewire components. A ShowEdit & ShowNotes component respectively.

This was working just fine until upgrading to version 3.

Post v3 upgrade, when the ShowNotes button is clicked, a modal allowing the user to view & create notes appears. Then when the ShowEdit button is clicked a different modal appears where the user can use it to edit the record.

The error appears after you close the ShowEdit modal and click the ShowNotes button, its dispatch event triggers just fine, however the ShowNotes modal will not appear. Important note, the modals are identical in nearly everyway safe for what makes them unique. How they open, their settings, are the same.

**Comment by mcpuishor** [comment]
Would be good if you could post some part of the code to better understand.

**Reply by Jeremiah-salley** [reply]
![ea34c3da-1448-4930-9013-6fb618e87ee7](https://github.com/livewire/livewire/assets/110695311/a67884c8-1adc-4c26-a027-a8cfce21988e)

![image](https://github.com/livewire/livewire/assets/110695311/a8a93635-eeb8-40ec-8ad9-6b7a75b4d9fa)


![5dbb990d-9c68-40f4-b86c-4b6ecd9191c5](https://github.com/livewire/livewire/assets/110695311/6f6f45c0-1b9a-4832-bdf3-5d82eed7dfb0)

![0f7c7487-8a19-4923-b319-fe995cbbbbfb](https://github.com/livewire/livewire/assets/110695311/14a910de-baff-4c1a-9329-8df472d1a699)


![3498a6e0-d9a9-4ddc-8e19-2cadf54c69da](https://github.com/livewire/livewire/assets/110695311/7f858323-fd39-48cf-a748-30f6a0a94b8f)


![65c9c2e8-8adf-495f-a04b-2a6b347eff55](https://github.com/livewire/livewire/assets/110695311/d5a7bad2-4d65-4d40-82ec-f46ed58e2df1)
